app/models/trigger_metadata_event_callback.rb

```diff
@@ -1,5 +1,7 @@
 class TriggerMetadataEventCallback
   def after_save(model)
+    return if environment(model).nil?
```

Comment:
we call this method twice, here and line 7... would be probably better not to do it twice...